### PR TITLE
PixelPaint: Add changing of a Guide to the context menu

### PR DESCRIPTION
**Pull request by TobyAsE**
This is something that annoys me a lot in Photoshop: Once you've added a Guide, you can drag it around but not (to my knowledge) change its numerical value or specify another relative measurement.
This PR implements this missing feature into PixelPaint. When hovering a Guide and opening the context menu you can now choose to change the Guide without dragging it around.
image
image

### CODE REVIEW

File: Userland/Applications/PixelPaint/EditGuideDialog.cpp [hunk 1]
```diff
@@ -62,4 +62,27 @@ EditGuideDialog::EditGuideDialog(GUI::Window* parent_window)
...
+    if (m_offset.ends_with('%')) {
+        auto percentage = m_offset.substring_view(0, m_offset.length() - 1).to_int();
+        if (!percentage.has_value())
+            return {};
+
+        if (orientation() == PixelPaint::Guide::Orientation::Horizontal)
+            offset = editor.image().size().height() * ((double)percentage.value() / 100.0);
+        else if (orientation() == PixelPaint::Guide::Orientation::Vertical)
+            offset = editor.image().size().width() * ((double)percentage.value() / 100.0);
+
```
Comment (alimpfard) [inline Userland/Applications/PixelPaint/EditGuideDialog.cpp:90]: Let's remove this line, we generally don't leave empty lines before `else`.

File: Userland/Applications/PixelPaint/Guide.h [hunk 2]
```diff
@@ -14,6 +14,7 @@ namespace PixelPaint {
 class Guide : public RefCounted<Guide> {
 public:
     enum class Orientation {
+        Unset,
```
Comment (alimpfard) [inline Userland/Applications/PixelPaint/Guide.h:17]: What does this value do, exactly?

**Review by alimpfard — COMMENTED** [review]
Looks pretty good, just some small comments.

File: Userland/Applications/PixelPaint/Guide.h (hunk 2 above)
Comment (TobyAsE) [reply Userland/Applications/PixelPaint/Guide.h:17]: It's passed to the EditGuideDialog as Orientation when creating a new Guide to not have one of the radio buttons pre-selected

Comment (alimpfard) [reply Userland/Applications/PixelPaint/Guide.h:17]: Hmm...okay. I guess that's fine.

**Review by alimpfard — APPROVED** [review]
Code LGTM.
I'm a bit too stuck in a regex yakstack to pull and take a look, so that might take a while unless someone else merges it :)

File: Userland/Applications/PixelPaint/Guide.h (hunk 2 above)
Comment (mustafaquraish) [reply Userland/Applications/PixelPaint/Guide.h:17]: Perhaps passing in an `Optional<Orientation>` would be better?. We can already compare `Optional<Orientation>` with `Orientation` types directly, so nothing else would really need to change, and you wouldn't need to have a separate `Unset` orientation value

**Pull request merged**